libusb/os/windows_winusb.c

```diff
@@ -1689,8 +1689,8 @@ static int winusb_get_device_list(struct libusb_context *ctx, struct discovered_
 				// https://docs.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regqueryvalueexa#remarks
 				// - "string may not have been stored with the proper terminating null characters"
 				// - "Note that REG_MULTI_SZ strings could have two terminating null characters"
-				if ((reg_type == REG_SZ && size >= sizeof(guid_string) - sizeof(char))
-				    || (reg_type == REG_MULTI_SZ && size >= sizeof(guid_string) - 2 * sizeof(char))) {
+				if ((reg_type == REG_SZ && size <= sizeof(guid_string) && guid_string[size-1] == 0)
+					|| (reg_type == REG_MULTI_SZ && size <= sizeof(guid_string) && guid_string[size-1] == 0 && guid_string[size-2] == 0)) {
```

Comment:
We usually have spaces around arithmetic operators "size - 1". And when checking for null characters the character '\0' constant is used for readability.

Reply:
resolved as you suggested